app/views/devise/sessions/new.html.erb

```diff
@@ -11,39 +11,18 @@
 </div>
  
 <div class="header_image">
-  
 </div>  
 
 <div class="log_in_box">
-  
-  <div class="box_tab">
-    LOG IN
-  </div>
-
-  <div class="form_box">
-    <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
-      <div class="">
-        <%= f.email_field :email, autofocus: true, placeholder: "Username or Email" %>
-      </div>
-
-      <div >
-        <%= f.password_field :password, autocomplete: "off", placeholder: "Password"  %>
-      </div>
-
-      <div class="form_buttons">
-        <div class="remember_button">
-        <% if devise_mapping.rememberable? -%>
-            <%= f.check_box :remember_me %>
-            <%= f.label :remember_me %>
-        <% end -%>
-        </div>
-        <div class="form_buttons">
-          <%= f.submit("Log in", class: 'selectButton') %>
-        </div>
-      </div>
+  <% if devise_mapping.omniauthable? %>
+    <% resource_class.omniauth_providers.each do |provider| %>
+      <%
+        display_name = OmniAuth::Utils.camelize(provider);
```

Comment:
what is a 'provider' here, and how many do we expect?
Either way, this type of code should be done in a view helper.

Reply:
The provider in this case is "urs" or "saml", it is basically the omniauth strategy name. I'll talk to you about view helper and purpose.

Reply:
I've included a helper method for the business logic that deals with camelizing the string.